vendor/k8s.io/kubernetes/pkg/controller/daemon/daemon_controller.go

```diff
@@ -1311,6 +1318,14 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 			}
 		}
 	}
+
+	if matches, matchErr := dsc.namespaceNodeSelectorMatches(node, ds); matchErr != nil {
+		err = matchErr
```

Comment:
return the error immediatelly, someone can add more code in upstream after that and this error can get silently ignored